plugins/outputs/elasticsearch/elasticsearch.go

```diff
@@ -0,0 +1,176 @@
+  server_host = "http://10.10.10.10:19200" # required.
+  index_name = "test" # required.
+  enable_sniffer = false
+  delimiter = "_"
+`
+
+type TimeSeries struct {
+	Series []*Metric `json:"series"`
+}
+
+type Metric struct {
+	Metric string   `json:"metric"`
+	Points [1]Point `json:"metrics"`
+}
+
+type Point [2]float64
+
+func (a *Elasticsearch) Connect() error {
+	if a.ServerHost == "" || a.IndexName == "" {
+		return fmt.Errorf("server_host and index_name are required fields for elasticsearch output")
+	}
+
+	client, err := elastic.NewClient(
+		elastic.SetHealthcheck(true),
+		elastic.SetSniff(a.EnableSniffer),
+		elastic.SetHealthcheckInterval(30*time.Second),
+		elastic.SetURL(a.ServerHost),
+	)
+
+	if err != nil {
+		return fmt.Errorf("FAILED to connect to elasticsearch host %s : %s\n", a.ServerHost, err)
+	}
+
+	a.Client = client
+
+	version, errVersion := a.Client.ElasticsearchVersion(a.ServerHost)
+
+	if errVersion != nil {
+		return fmt.Errorf("FAILED to get elasticsearch version : %s\n", errVersion)
+	}
+
+	a.Version = version
+
+	return nil
+}
+
+func (a *Elasticsearch) Write(metrics []telegraf.Metric) error {
+	if len(metrics) == 0 {
+		return nil
+	}
+
+	for _, metric := range metrics {
+
+		m := make(map[string]interface{})
+		//m["created"] = metric.UnixNano() / 1000000
+		m["created"] = time.Now()
+
+		if host, ok := metric.Tags()["host"]; ok {
+			m["host"] = host
+		} else {
+			host, err := os.Hostname()
+			if err != nil {
+				panic(err)
+			}
+			m["host"] = host
+		}
+
+		// Earlier versions of EL doesnt accept '.' in field name
+		if len(a.Separator) > 1 {
+			return fmt.Errorf("FAILED Separator exceed one character : %s\n", a.Separator)
+		}
+
+		for key, value := range metric.Tags() {
+			if key != "host" {
+				if strings.HasPrefix(a.Version, "2.") {
+					m[strings.Replace(key, ".", a.Separator, -1)] = value
+				} else {
+					m[key] = value
+				}
+			}
+		}
+
+		for key, value := range metric.Fields() {
+			if strings.HasPrefix(a.Version, "2.") {
+				m[strings.Replace(key, ".", a.Separator, -1)] = value
+			} else {
+				m[key] = value
+			}
+		}
+
+		_, errMessage := a.Client.Index().
+			Index(a.IndexName).
+			Type(metric.Name()).
+			BodyJson(m).
+			Do()
+
+		if errMessage != nil {
+			return fmt.Errorf("FAILED to send elasticsearch message to index %s : %s\n", a.IndexName, errMessage)
+		}
+
+	}
+
+	return nil
+}
+
+func (a *Elasticsearch) SampleConfig() string {
+	return sampleConfig
+}
+
+func (a *Elasticsearch) Description() string {
+	return "Configuration for Elasticsearch to send metrics to."
+}
+
+func buildMetrics(m telegraf.Metric) (map[string]Point, error) {
+	ms := make(map[string]Point)
+	for k, v := range m.Fields() {
+		var p Point
+		if err := p.setValue(v); err != nil {
+			return ms, fmt.Errorf("unable to extract value from Fields, %s", err.Error())
+		}
+		p[0] = float64(m.Time().Unix())
+		ms[k] = p
+	}
+	return ms, nil
+}
+
+func (p *Point) setValue(v interface{}) error {
```

Comment:
Changed in commit:
https://github.com/influxdata/telegraf/pull/1875/commits/862f728635110c159444d2a88bdf3cdf9ae47edc